use geojson endpoint, reverse coordinate order

jaredkosanovic opened the pull request:
One of the changes includes using a geojson specific endpoint which, depending on if the geometry is polygon or multipolygon, could have coordinates of a 3 or 4 dimensional array, respectively. I used https://github.com/paulmach/go.geojson to take care of unmarshaling a json body into a featurecollection while using a 3 or 4 dimensional array appropriately

The arcgis geojson endpoint isn't adhering to RFC 7946 standards regarding the right-hand wrap protocol. Specifically, for a given polygon, the first/exerior ring must wrap in a counterclockwise direction while all rings following the first/exterior ring must wrap in a clockwise direction.

Arcgis data was doing the opposite, so I created two data slices containing arcgis data, and reassigned the values for one of them to reverse the direction that the coordinates were layed out in a ring. The order of the rings themselves within a polygon should remain the same, since arcgis is correctly putting the exterior ring first with all interior rings following the first ring.

jaredkosanovic committed:
use geojson endpoint, reverse coordinate order

One of the changes includes using a geojson specific endpoint which, depending on if the geometry is polygon or multipolygon, could have coordinates of a 3 or 4 dimensional array, respectively. I used https://github.com/paulmach/go.geojson to take care of unmarshaling a json body into a featurecollection while using a 3 or 4 dimensional array appropriately

The arcgis geojson endpoint isn't adhering to RFC 7946 standards regarding the right-hand wrap protocol. Specifically, for a given polygon, the first/exerior ring must wrap in a counterclockwise direction while all rings following the first/exterior ring must wrap in a clockwise direction.

Arcgis data was doing the opposite, so I created two data slices containing arcgis data, and reassigned the values for one of them to reverse the direction that the coordinates were layed out in a ring. The order of the rings themselves within a polygon should remain the same, since arcgis is correctly putting the exterior ring first with all interior rings following the first ring.

jaredkosanovic commented:
Note, at this time, the code is a work in progress since it doesn't handle multipolygons. This should be an easy addition since it just adds one more level deep of recursion.

floppyears commented on converter.go:
```diff
@@ -85,26 +90,30 @@ func convertCoordinates(lon, lat float64) (float64, float64) {
-                buildings.Features[featureIndex].Geometry.Rings[ringIndex][coordPairIndex][0] = convertedLon
-                buildings.Features[featureIndex].Geometry.Rings[ringIndex][coordPairIndex][1] = convertedLat
+        fmt.Println("Converting building", building.Properties["BldNam"])
+        if building.Geometry.Type == "Polygon" {
```
You may want to refactor some of the logic in this  for loop since we'll have to deal with different data types in the future (e.g.: blue phones are points)

jaredkosanovic replied:
I could, but I'd rather have a separate script for converting those coordinates. I can always move the cs2cs logic into a separate package. I think it would be easier to keep this program limited to pure geojson

floppyears replied:
Fair enough, we can postpone the code refactoring until we have to handle more data points or different data types.

jaredkosanovic committed:
handle multipolygons

jaredkosanovic committed:
remove offset

I realized all the buildings were shifted slightly to the west. I removed this offset because I don't know why it was there in the first place. See JIRA CO-816 for more details

floppyears commented on Dockerfile:
```diff
@@ -0,0 +1,3 @@
+FROM golang:1.8.1
+
+RUN go get github.com/paulmach/go.geojson
```
How about if we mention in the go.geojson as a dependency in the README?

jaredkosanovic committed:
add geojson library dependency to readme

jaredkosanovic committed:
make it a link
